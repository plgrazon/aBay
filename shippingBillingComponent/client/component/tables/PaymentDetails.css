:local(.payment) {
  border-collapse: collapse;
  font-size: 16px;
  font-family: helvetica, arial, sans-serif;
  width: 98%;
  text-align: left;
  margin-left: auto;
  margin-right: auto;
}

:local(.payment-outside-border) {
  border: 1px solid black;
}

:local(.payment-header) {
  border-bottom: 1px solid black;
  border-collapse: collapse;
  background-color: #e8eaed;
}

:local(th) {
  height: 40px;
  text-align: left;
}

:local(td) {
  height: 30px;
  text-align: left;
}

:local(.paypal) {
  max-width: 100%;
  height: 50px;
}
