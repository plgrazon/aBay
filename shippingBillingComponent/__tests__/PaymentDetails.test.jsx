import PaymentDetails from '../client/component/tables/PaymentDetailsTable';

describe('Component: PaymentDetailsTable', () => {
  const wrapper = shallow(<PaymentDetails />);

  it('PaymentDetailsTable should be defined', () => {
    expect(wrapper).toBeDefined();
  });

  it('PaymentDetailsTable should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('PaymentDetailsTable should render table', () => {
    expect(wrapper.find('table').length).toBe(1);
  });
});
