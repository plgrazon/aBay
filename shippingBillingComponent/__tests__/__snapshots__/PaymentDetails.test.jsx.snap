// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: PaymentDetailsTable PaymentDetailsTable should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PaymentDetails />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Payment details
        </h3>,
        <table>
          <thead>
            <tr>
              <th>
                Payment Method
              </th>
              <th>
                Preferred / Accepted
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                <img
                  src="https://s3.us-east-2.amazonaws.com/shippingbillingcomponent/paypalOptions.jpeg"
                />
              </td>
              <td>
                PayPal Preferred
              </td>
            </tr>
          </tbody>
        </table>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Payment details",
        },
        "ref": null,
        "rendered": "Payment details",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <thead>
              <tr>
                <th>
                  Payment Method
                </th>
                <th>
                  Preferred / Accepted
                </th>
              </tr>
            </thead>,
            <tbody>
              <tr>
                <td>
                  <img
                    src="https://s3.us-east-2.amazonaws.com/shippingbillingcomponent/paypalOptions.jpeg"
                  />
                </td>
                <td>
                  PayPal Preferred
                </td>
              </tr>
            </tbody>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr>
                <th>
                  Payment Method
                </th>
                <th>
                  Preferred / Accepted
                </th>
              </tr>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <th>
                    Payment Method
                  </th>,
                  <th>
                    Preferred / Accepted
                  </th>,
                ],
                "className": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Payment Method",
                  },
                  "ref": null,
                  "rendered": "Payment Method",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Preferred / Accepted",
                  },
                  "ref": null,
                  "rendered": "Preferred / Accepted",
                  "type": "th",
                },
              ],
              "type": "tr",
            },
            "type": "thead",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr>
                <td>
                  <img
                    src="https://s3.us-east-2.amazonaws.com/shippingbillingcomponent/paypalOptions.jpeg"
                  />
                </td>
                <td>
                  PayPal Preferred
                </td>
              </tr>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <td>
                    <img
                      src="https://s3.us-east-2.amazonaws.com/shippingbillingcomponent/paypalOptions.jpeg"
                    />
                  </td>,
                  <td>
                    PayPal Preferred
                  </td>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      src="https://s3.us-east-2.amazonaws.com/shippingbillingcomponent/paypalOptions.jpeg"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": undefined,
                      "src": "https://s3.us-east-2.amazonaws.com/shippingbillingcomponent/paypalOptions.jpeg",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "PayPal Preferred",
                  },
                  "ref": null,
                  "rendered": "PayPal Preferred",
                  "type": "td",
                },
              ],
              "type": "tr",
            },
            "type": "tbody",
          },
        ],
        "type": "table",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Payment details
          </h3>,
          <table>
            <thead>
              <tr>
                <th>
                  Payment Method
                </th>
                <th>
                  Preferred / Accepted
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  <img
                    src="https://s3.us-east-2.amazonaws.com/shippingbillingcomponent/paypalOptions.jpeg"
                  />
                </td>
                <td>
                  PayPal Preferred
                </td>
              </tr>
            </tbody>
          </table>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Payment details",
          },
          "ref": null,
          "rendered": "Payment details",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <thead>
                <tr>
                  <th>
                    Payment Method
                  </th>
                  <th>
                    Preferred / Accepted
                  </th>
                </tr>
              </thead>,
              <tbody>
                <tr>
                  <td>
                    <img
                      src="https://s3.us-east-2.amazonaws.com/shippingbillingcomponent/paypalOptions.jpeg"
                    />
                  </td>
                  <td>
                    PayPal Preferred
                  </td>
                </tr>
              </tbody>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <tr>
                  <th>
                    Payment Method
                  </th>
                  <th>
                    Preferred / Accepted
                  </th>
                </tr>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <th>
                      Payment Method
                    </th>,
                    <th>
                      Preferred / Accepted
                    </th>,
                  ],
                  "className": undefined,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Payment Method",
                    },
                    "ref": null,
                    "rendered": "Payment Method",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Preferred / Accepted",
                    },
                    "ref": null,
                    "rendered": "Preferred / Accepted",
                    "type": "th",
                  },
                ],
                "type": "tr",
              },
              "type": "thead",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <tr>
                  <td>
                    <img
                      src="https://s3.us-east-2.amazonaws.com/shippingbillingcomponent/paypalOptions.jpeg"
                    />
                  </td>
                  <td>
                    PayPal Preferred
                  </td>
                </tr>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <td>
                      <img
                        src="https://s3.us-east-2.amazonaws.com/shippingbillingcomponent/paypalOptions.jpeg"
                      />
                    </td>,
                    <td>
                      PayPal Preferred
                    </td>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <img
                        src="https://s3.us-east-2.amazonaws.com/shippingbillingcomponent/paypalOptions.jpeg"
                      />,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": undefined,
                        "src": "https://s3.us-east-2.amazonaws.com/shippingbillingcomponent/paypalOptions.jpeg",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "PayPal Preferred",
                    },
                    "ref": null,
                    "rendered": "PayPal Preferred",
                    "type": "td",
                  },
                ],
                "type": "tr",
              },
              "type": "tbody",
            },
          ],
          "type": "table",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
